docs/rest/CompartmentSearchExamples.http

```diff
@@ -0,0 +1,901 @@
+                },
+                {
+                  "category": {
+                    "coding": [
+                      {
+                        "code": "benefit"
+                      }
+                    ]
+                  },
+                  "amount": {
+                    "value": 180.00,
+                    "currency": "USD"
+                  }
+                }
+              ]
+            }
+          ]
+        }
+      ]
+    }
+  ],
+  "total": [
+    {
+      "category": {
+        "coding": [
+          {
+            "code": "submitted"
+          }
+        ]
+      },
+      "amount": {
+        "value": 135.57,
+        "currency": "USD"
+      }
+    },
+    {
+      "category": {
+        "coding": [
+          {
+            "code": "benefit"
+          }
+        ]
+      },
+      "amount": {
+        "value": 96.00,
+        "currency": "USD"
+      }
+    }
+  ]
+}
+
+###
+# Create a procedure that references the patient with the id "patient".
+# This will be in the patient's compartment.
+PUT https://{{hostname}}/Procedure/appendectomy
+Content-Type: application/json
+Authorization: Bearer {{bearer.response.body.access_token}}
+
+{
+  "resourceType": "Procedure",
+  "id": "appendectomy",
+  "text": {
+    "status": "generated",
+    "div": "<div xmlns=\"http://www.w3.org/1999/xhtml\">Routine Appendectomy</div>"
+  },
+  "status": "completed",
+  "code": {
+    "coding": [
+      {
+        "system": "http://snomed.info/sct",
+        "code": "80146002",
+        "display": "Appendectomy (Procedure)"
+      }
+    ],
+    "text": "Appendectomy"
+  },
+  "subject": {
+    "reference": "Patient/patient"
+  },
+  "recorder": {
+    "reference": "Practitioner/dude",
+    "display": "Dr Adam Careful"
+  },
+  "performer": [
+    {
+      "actor": {
+        "reference": "Practitioner/dude",
+        "display": "Dr Adam Careful"
+      }
+    }
+  ],
+  "followUp": [
+    {
+      "text": "ROS 5 days  - 2013-04-10"
+    }
+  ],
+  "note": [
+    {
+      "text": "Routine Appendectomy. Appendix was inflamed and in retro-caecal position"
+    }
+  ]
+}
+
+###
+# Run patient everything, but filter by explanation of benefit
+# resources.
+# This should return the two explanation of benefit resources
+# in the patient's compartment.
+GET {{hostname}}/Patient/patient/$everything?_type=ExplanationOfBenefit
+Authorization: Bearer {{bearer.response.body.access_token}}
+
+###
+# Get everything from the patient's compartment.
+# This should return two explanations of benefits and one procedure.
+GET {{hostname}}/Patient/patient/*
+Authorization: Bearer {{bearer.response.body.access_token}}
+
+###
+# Get the patient's compartment, but filter for explanation of benefit
+# and observation resources.
+# This should return two explanation of benefit resources.
+GET {{hostname}}/Patient/patient/*?_type=ExplanationOfBenefit,Observation
+Authorization: Bearer {{bearer.response.body.access_token}}
```

Comment:
To cover some more cases, did we have tests for these?

// Single compartment
GET {{hostname}}/Patient/patient/ExplanationOfBenefit

//Mismatched types:
GET {{hostname}}/Patient/patient/ExplanationOfBenefit?_type=Observation